src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const KeyAndValues = sourceString.split(';');
+  const end = {};
+
+  KeyAndValues.forEach((element) => {
+    const preElement = element.split(':');
+
+    if (preElement[0]) {
+      preElement[0] = preElement[0].trim();
+    }
+
+    if (preElement[1]) {
+      preElement[1] = preElement[1].trim();
+    }
+
+    end[preElement[0]] = preElement[1];
```

Comment:
The code directly modifies an array element within a forEach loop, which is not recommended. Instead, you should create a new object inside the loop and assign the processed key-value pair to the result object.